### PR TITLE
proxylite: Add HTTPS backend support

### PR DESCRIPTION
**Pull request by nlsun**
This changes the internal route syntax.

### CODE REVIEW

**Comment by nlsun** [comment]
@nickbp @gabrielhartmann  Ready for review.

File: docs/pages/dev-guide/developer-guide.md [hunk 1]
```diff
@@ -1366,15 +1366,15 @@ pods:
         env:
```
Comment (nickbp) [inline docs/pages/dev-guide/developer-guide.md]: Does the example's proxylite version need to be bumped to `2.0.0`?

Comment (nlsun) [inline docs/pages/dev-guide/developer-guide.md]: Ah, yes

**Comment by nlsun** [comment]
Need to rebase after this documentation PR goes through: https://github.com/mesosphere/dcos-commons/pull/600

**Comment by nlsun** [comment]
@mesosphere-ci Retest this please.

**Comment by mohitsoni** [comment]
retest this please

**Comment by nlsun** [comment]
@mohitsoni @nickbp @gabrielhartmann can someone merge?

**Pull request merged**